File: public/api/panel/configuracoes.php
<?php
ini_set('display_errors', 0);
error_reporting(0);

// Carrega o autoloader do Composer
require_once __DIR__ . '/../../../vendor/autoload.php';

// Carrega as variáveis de ambiente
$envPath = __DIR__ . '/../../../.env';
if (file_exists($envPath)) {
    $dotenv = Dotenv\Dotenv::createImmutable(__DIR__ . '/../../../');
    $dotenv->load();
}

// Usa a classe Database para conexão
use App\Config\Database;

try {
    $pdo = Database::getConnection();
} catch (PDOException $e) {
    http_response_code(500);
    echo json_encode(['error' => 'Erro na conexão com banco de dados']);
    exit;
}

header('Content-Type: application/json');
header('Access-Control-Allow-Origin: *');
header('Access-Control-Allow-Methods: GET, POST, OPTIONS');
header('Access-Control-Allow-Headers: Content-Type');

if ($_SERVER['REQUEST_METHOD'] === 'OPTIONS') {
    exit(0);
}

try {
    if ($_SERVER['REQUEST_METHOD'] === 'GET') {
        // Buscar todas as configurações
        $stmt = $pdo->prepare("SELECT chave, valor FROM configuracoes ORDER BY id");
        $stmt->execute();
        $configuracoes = $stmt->fetchAll(PDO::FETCH_ASSOC);
        
        echo json_encode([
            'success' => true,
            'configuracoes' => $configuracoes
        ]);
        
    } elseif ($_SERVER['REQUEST_METHOD'] === 'POST') {
        // Atualizar configurações
        $input = json_decode(file_get_contents('php://input'), true);
        
        if (!$input) {
            throw new Exception('Dados inválidos');
        }
        
        $pdo->beginTransaction();
        
        foreach ($input as $chave => $valor) {
            $stmt = $pdo->prepare("UPDATE configuracoes SET valor = ?, updated_at = NOW() WHERE chave = ?");
            $stmt->execute([$valor, $chave]);
        }
        
        $pdo->commit();
        
        echo json_encode([
            'success' => true,
            'message' => 'Configurações atualizadas com sucesso'
        ]);
        
    } else {
        throw new Exception('Método não permitido');
    }
    
} catch (Exception $e) {
    if (isset($pdo)) {
        $pdo->rollBack();
    }
    
    echo json_encode([
        'success' => false,
        'message' => $e->getMessage()
    ]);
} 

File: app/routers/panel/configuracoes.py
from fastapi import APIRouter, Depends, HTTPException
from fastapi.responses import JSONResponse
from ...core.db import get_db

router = APIRouter(prefix="/panel", tags=["panel-configuracoes"])

@router.get("/configuracoes")
async def listar_configuracoes(db = Depends(get_db)):
    try:
        with db.cursor() as cur:
            cur.execute("SELECT chave, valor FROM configuracoes ORDER BY id")
            configuracoes = cur.fetchall()
            return JSONResponse(content={"success": True, "configuracoes": configuracoes})
    except Exception as e:
        return JSONResponse(content={"error": "Erro na conexão com banco de dados"}, status_code=500)

@router.post("/configuracoes")
async def atualizar_configuracoes(payload: dict, db = Depends(get_db)):
    if not payload:
        raise HTTPException(status_code=400, detail="Dados inválidos")
    try:
        with db.cursor() as cur:
            for chave, valor in payload.items():
                cur.execute(
                    "UPDATE configuracoes SET valor = %s, updated_at = NOW() WHERE chave = %s",
                    (valor, chave)
                )
        try:
            db.commit()
        except Exception:
            pass
        return JSONResponse(content={"success": True, "message": "Configurações atualizadas com sucesso"})
    except Exception:
        try:
            db.rollback()
        except Exception:
            pass
        return JSONResponse(content={"success": False, "message": "Erro ao atualizar configurações"})

